[9.4] Document Agent Builder workflow management capabilities (#6216)
elastic/docs-content: explore-analyze/ai-features/agent-builder/builtin-skills-reference.md

$$$agent-builder-streams-exploration-skill$$$ `streams-exploration` {applies_to}`stack: ga 9.4+`
:   Discovers, inspects, and queries {{es}} streams. Use when a user wants to list available streams, understand a stream's schema, check data quality or retention, or sample documents from a stream. This is a read-only skill: it cannot create, update, or delete streams or modify stream configuration.

$$$agent-builder-workflow-authoring-skill$$$ `workflow-authoring` {applies_to}`stack: preview 9.4` {applies_to}`serverless: preview`

> :   Creates, modifies, and validates [Elastic Workflows](/explore-analyze/workflows.md) YAML definitions from natural language. Covers step types, triggers, Liquid templating, connector integrations, and validation. Use when a user wants to draft a new workflow from a description, find and load an existing workflow for editing, add, replace, or delete steps, change a workflow's trigger, or update top-level properties such as name, description, or tags. The agent validates the generated or modified YAML before proposing the change so the user can accept or decline it.

Reviewer: ```suggestion
:   Creates, modifies, and validates [Elastic Workflows](/explore-analyze/workflows.md) YAML definitions from natural language user input. Covers step types, triggers, Liquid templating, connector integrations, and validation. Use this skill when a user wants to draft a new workflow from a description, edit an existing workflow, change a workflow's trigger, or update top-level properties such as name, description, or tags. The agent validates the generated or modified YAML before proposing the change so the user can accept or decline it.
```

minor language tweaks to avoid going overboard with details, lmkwyt

    **Assigned tools:**

    Lookup: `platform.workflows.get_step_definitions`, `platform.workflows.get_trigger_definitions`, `platform.workflows.get_examples`, `platform.workflows.get_connectors`, `platform.workflows.validate_workflow`

    Edit: `platform.workflows.workflow_set_yaml`, `platform.workflows.workflow_insert_step`, `platform.workflows.workflow_modify_step`, `platform.workflows.workflow_modify_step_property`, `platform.workflows.workflow_modify_property`, `platform.workflows.workflow_delete_step`

>     To discover existing workflows, the skill uses the platform's Semantic Metadata Layer (SML) tools (`platform.core.sml_search` to find workflows by name, description, or tags, and `platform.core.sml_attach` to load a workflow into the conversation as a `workflow.yaml` attachment for editing).

Reviewer: as previous comment, let's not mention the SML which is implementation details
